server/src/main/java/org/apache/iotdb/db/engine/settle/SettleRequestHandler.java

```diff
@@ -0,0 +1,228 @@
+        return RpcUtils.getStatus(
+            TSStatusCode.ILLEGAL_PATH, "StorageGroup of files is not consistent.");
+      }
+
+      long timePartitionOfFile = TsFileUtils.getTimePartition(tsFile);
+      if (timePartitionId == null) {
+        timePartitionId = timePartitionOfFile;
+      } else if (timePartitionId != timePartitionOfFile) {
+        return RpcUtils.getStatus(
+            TSStatusCode.ILLEGAL_PATH, "TimePartition of files is not consistent.");
+      }
+
+      String fileName = tsFile.getName();
+      TsFileNameGenerator.TsFileName tsFileName;
+      try {
+        tsFileName = TsFileNameGenerator.getTsFileName(fileName);
+      } catch (IOException e) {
+        return RpcUtils.getStatus(TSStatusCode.ILLEGAL_PATH, "Meet error when parsing TsFileName.");
+      }
+
+      int levelOfFile = tsFileName.getInnerCompactionCnt();
+      if (level == null) {
+        level = levelOfFile;
+      } else if (level != levelOfFile) {
+        return RpcUtils.getStatus(
+            TSStatusCode.ILLEGAL_PARAMETER, "Level of files is not consistent.");
+      }
+
+      if (TsFileUtils.isSequence(tsFile)) {
+        hasSeqFile = true;
+      } else {
+        hasUnSeqFile = true;
+      }
+      tsFileNames.add(fileName);
+      if (hasSeqFile && hasUnSeqFile) {
+        return RpcUtils.getStatus(
+            TSStatusCode.UNSUPPORTED_OPERATION, "Settle by cross compaction is not allowed.");
+      }
+    }
+
+    if (!hasModsFile) {
+      return RpcUtils.getStatus(
+          TSStatusCode.ILLEGAL_PARAMETER, "At least one mods file should be selected");
+    }
+    if (dataRegion == null) {
+      return RpcUtils.getStatus(TSStatusCode.ILLEGAL_PATH, "DataRegion not exist");
+    }
+    TsFileManager tsFileManager = dataRegion.getTsFileManager();
+    if (!tsFileManager.isAllowCompaction()) {
+      return RpcUtils.getStatus(
+          TSStatusCode.UNSUPPORTED_OPERATION, "Compaction in this DataRegion is not allowed.");
+    }
+
+    // try to get a continuous TsFileResource list with input TsFile names
+    List<TsFileResource> tsFileResources =
+        getContinuousTsFileResourcesByFileNames(
+            tsFileManager, timePartitionId, hasSeqFile, tsFileNames);
+    if (tsFileResources.size() != tsFileNames.size()) {
+      return RpcUtils.getStatus(
+          TSStatusCode.ILLEGAL_PARAMETER, "Could not find enough satisfied TsFile");
+    }
+
+    IoTDBConfig config = IoTDBDescriptor.getInstance().getConfig();
+    if (hasSeqFile && !config.isEnableSeqSpaceCompaction()) {
+      return RpcUtils.getStatus(
+          TSStatusCode.UNSUPPORTED_OPERATION, "Compaction in Seq Space is not enabled");
+    }
+    if (hasUnSeqFile && !config.isEnableUnseqSpaceCompaction()) {
+      return RpcUtils.getStatus(
+          TSStatusCode.UNSUPPORTED_OPERATION, "Compaction in Unseq Space is not enabled");
+    }
+    if (tsFileResources.size() > config.getMaxInnerCompactionCandidateFileNum()) {
```

Comment:
We can move this validation before the continuous file validation because this check is more simple to do, which will reduce the resource usage in some scenarios.